bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/checksum/CRC32CDigestManager.java

```diff
@@ -41,8 +43,14 @@ protected MutableInt initialValue() throws Exception {
 
     public CRC32CDigestManager(long ledgerId, boolean useV2Protocol, ByteBufAllocator allocator) {
         super(ledgerId, useV2Protocol, allocator);
+
+        if (!isSupported) {
```

Comment:
I don't think `isSupported` is a good name. it is actually a confusing name in this case. because the flag isn't used for telling whether CRC32C is support. The flag should be used for controlling whether the error log message is printed or not.

I would suggest renaming it to "nonSupportedMessagePrinted" and make the default value as `false`.

Then you can change the logic to:

```
if (!Sse42Crc32C.isSupported() && !nonSupportedMessagePrinted) {
    log.error("Sse42Crc32C is not supported, will use a slower CRC32C implementation.");
    nonSupportedMessagePrinted = true;
}
```